clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -2264,7 +2264,9 @@ CIRGenModule::createCIRFunction(mlir::Location loc, StringRef name,
     mlir::SymbolTable::setSymbolVisibility(
         f, mlir::SymbolTable::Visibility::Private);
 
-    setExtraAttributesForFunc(f, FD);
+    // Initialize with empty dict of extra attributes.
+    f.setExtraAttrsAttr(mlir::cir::ExtraFuncAttributesAttr::get(
+        builder.getContext(), builder.getDictionaryAttr({})));
```

Comment:
Initialize it to ensure existing chain calls like `func.getExtraFuncAttrs().getElements()` do not fail.